CMakeLists.txt

```
@@ -132,6 +132,28 @@ option(BUILD_DOCS "Build Api documentation")
 cvc5_option(STATIC_BINARY "Link against static system libraries \
 (enabled by default for static builds)")
 
+# Custom flags for WebAssembly compilation
+set(WASM_FLAGS "" CACHE STRING "Link flags for the WebAssembly binary generation")
+set(WASM "OFF" CACHE STRING "Use a specific extension for WebAssembly compilation")
+
+# Make sure no other avaible value will be assigned to this variable
+if(NOT(WASM STREQUAL "WASM" OR WASM STREQUAL "JS" OR WASM STREQUAL "HTML" OR WASM STREQUAL "OFF"))
+  message(FATAL_ERROR "${WASM} is not a valid wasm flag value.")
+endif()
```

You should be able to define the set of possible values as follows (untested):
```suggestion
set_property(CACHE WASM PROPERTY STRINGS WASM JS HTML OFF)
```

---

I tested and set_property apparently doesn't execute tests to check if the variable is in the desired domain. One way to approach that way is something like that:

```
set_property(CACHE WASM PROPERTY STRINGS WASM JS HTML OFF)
get_property(WASM_STRINGS CACHE WASM PROPERTY STRINGS)

# Make sure no other avaible value will be assigned to this variable
if(NOT WASM IN_LIST WASM_STRINGS)
  message(FATAL_ERROR "${WASM} is not a valid wasm flag value.")
endif()
```